### PR TITLE
Issue/941 Add `recrawl` API endpoints to sleuther framework

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Added two APIs to sleuther framework:
- POST /recrawl
  - make sleuther recrawl the registry and respond `crawler started`
  - If it's crawling, `in-progress` will be responded
- GET /crawlerProgress
  - return a JSON like this:

```
{"isCrawling":true,"crawlingPageToken":"932","crawledRecordNumber":200}
```

### Checklist
- [x] There are unit tests to verify my changes are correct
- [x] I've updated CHANGES.md with what I changed.
- [x] I've linked this PR to an issue in ZenHub (or there is no issue) and dragged it to the _Pull Request_ column

### CODE REVIEW

File: magda-sleuther-framework/src/setupRecrawlEndpoint.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    server: Application,
+    options: SleutherOptions,
+    recrawlFunc: () => Promise<void>,
+    crawlerProgressFunc: () => {
+        isCrawling: boolean;
+        crawlingPageToken: string;
+        crawledRecordNumber: number;
+    }
+) {
+    server.get("/recrawl", (request, response) => {
```
Comment (AlexGilleran) [inline magda-sleuther-framework/src/setupRecrawlEndpoint.ts]: Can we make this a `POST` with no body instead? `GET` should be idempotent :)

File: magda-sleuther-framework/src/index.ts [hunk 2]
```diff
@@ -9,10 +9,26 @@ import AsyncPage, {
...
 import setupWebhookEndpoint from "./setupWebhookEndpoint";
+import setupRecrawlEndpoint from "./setupRecrawlEndpoint";
+import startApiEndpoints from "./startApiEndpoints";
 import isWebhookRegistered from "./isWebhookRegistered";
 import registerWebhook from "./registerWebhook";
 import resumeWebhook from "./resumeWebhook";
 
+let isCrawling: boolean = false;
+let crawlingPageToken: string = "";
+let crawledRecordNumber: number = 0;
```
Comment (AlexGilleran) [inline magda-sleuther-framework/src/index.ts]: Could we lift all this crawler state stuff into its own class instead of having it in `index.ts` ?

**Comment by t83714** [comment]
@AlexGilleran 
Test cases are added.
It's good to review again~ Thank you!

File: magda-sleuther-framework/src/test/Crawler.spec.ts [hunk 3]
```diff
@@ -0,0 +1,315 @@
+import { Server } from "http";
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-sleuther-framework/src/test/Crawler.spec.ts]: has re-written the code

**Comment by t83714** [comment]
@AlexGilleran 
It's ready to re-look at again~
I've cleaned up the code a little bit.
Thanks to the property test. I also discovered a random issue with progress API (that test case couldn't pass every time).
It's now has been fixed as well.

**Pull request merged**